src/App.tsx

```diff
@@ -1,61 +1,142 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { ChangeEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todos } from './components/TodoInfo';
+
+function getUserById(userId: number) {
+  return usersFromServer.find((user) => user.id === userId);
+}
+
+export const todos = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export function getMaxId(todo: Todos[]) {
+  const maxId = Math.max(0, ...todo.map((item) => item.id));
+
+  return maxId + 1;
+}
+
+const initialUserState = {
+  id: 0,
+  title: '',
+  userId: 0,
+  completed: false,
+};
 
 export const App = () => {
+  const [todo, setTodo] = useState(initialUserState);
+
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const [newTodo, setNewTodo] = useState<Todos[]>(todos);
```

Comment:
your variable called 'newTodo', but at the beginning it has initial value as todos array, you should pick a better name or rethink this logic